### PR TITLE
chore(#9327): add feature flag and hide old navigation

### PR DESCRIPTION
**Pull request by Benmuiruri**
# Description

This is the first deliverable of this issue to update the [Navigation](https://github.com/medic/care-teams/issues/24)

This PR completes this [task](https://github.com/medic/care-teams/issues/127)  to Add feature flag and hide old navigation
- Add new permission in default config with no roles: `can_view_old_navigation`
- Add a constant for this permission `header.component.ts` and exports it.
- hides the old navigation by default and only shows old navigation if user has permission. 

<details>
<summary>Desktop view</summary>

Screenshot 2024-08-14 at 14 57 38

</details>

<details>
<summary>Mobile View  with search</summary>

Screenshot 2024-08-14 at 14 57 57

</details>

<details>
<summary>Mobile view without search</summary>

Screenshot 2024-08-14 at 14 58 12

</details>

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs
If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:feature-flag-hide-old-nav/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:feature-flag-hide-old-nav/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:feature-flag-hide-old-nav/docker-compose/cht-couchdb-clustered.yml)

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: webapp/src/ts/app.component.html [hunk 1]
```diff
@@ -14,12 +14,9 @@
 
   <mm-actionbar *ngIf="useOldActionBar" [reportForms]="reportForms"></mm-actionbar>
 
-  <div class="container-fluid" *ngIf="initialisationComplete">
-    <mm-header
-      [adminUrl]="adminUrl"
-      [canLogOut]="canLogOut"
-    ></mm-header>
-    
+  <div class="container-fluid" *ngIf="initialisationComplete" [class.old-nav]="!useOldNav">
```
Comment (latin-panda) [inline webapp/src/ts/app.component.html:17]: The `old-nav` class is applied when the auth.has(`can_view_old_navigation`) is true.

```suggestion
  <div class="container-fluid" *ngIf="initialisationComplete" [class.old-nav]="useOldNav">
```

**Review by latin-panda — CHANGES_REQUESTED** [review]
Thanks for shipping! 
One small comment in the html but you will need to update the css too

**Review by latin-panda — APPROVED** [review]
Looks good! ✨
Thanks @Benmuiruri , ready to go!

**Review by kapelo — APPROVED** [review]
LGTM!

**Pull request merged**